Source/JavaScriptCore/runtime/TemporalCalendar.cpp

```diff
@@ -291,6 +293,32 @@ static bool balanceISODate(double& year, double& month, double& day)
     return true;
 }
 
+static int epochTimeToEpochYear(double t)
+{
+    return msToYear(t);
+}
+
+static int32_t epochTimeToMonthInYear(double t)
+{
+    return std::get<1>(WTF::yearMonthDayFromDays(msToDays(t)));
+}
+
+static int32_t epochTimeToDate(double t)
+{
+    return std::get<2>(WTF::yearMonthDayFromDays(msToDays(t)));
+}
+
+// https://tc39.es/proposal-temporal/#sec-temporal-balanceisodate
+ISO8601::PlainDate TemporalCalendar::balanceISODate(double year, double month, double day)
+{
+    auto epochDays = makeDay(year, month - 1, day);
+    double ms = makeDate(epochDays, 0);
+    int32_t y = epochTimeToEpochYear(ms);
+    int32_t m = epochTimeToMonthInYear(ms) + 1;
```

Comment:
We should not call `WTF::yearMonthDayFromDays` directly since it is costly.
Please use JSDateMath's cache.

Reply:
Changed in [bd306a9](https://github.com/WebKit/WebKit/pull/51637/commits/bd306a96d53a53ed9d9c7d68f5b26a28447e5c27)